File: web/medistream/src/features/patientqueue/services/queueService.js
import API from '../../../services/api';

export const queueService = {
  joinQueue: async (queueRequestData) => {
    try {
      const response = await API.post('/api/queue/join', queueRequestData);
      return response.data;
    } catch (error) {
      console.error('Error joining queue:', error);
      throw error;
    }
  },

  getAllQueue: async () => {
    try {
      const response = await API.get('/api/queue');
      return response.data;
    } catch (error) {
      console.error('Error fetching queue:', error);
      throw error;
    }
  },

  updateQueueItem: async (id, queueData) => {
    try {
      const response = await API.put(`/api/queue/${id}`, queueData);
      return response.data;
    } catch (error) {
      console.error('Error updating queue item:', error);
      throw error;
    }
  },

  deleteQueueItem: async (id) => {
    try {
      const response = await API.delete(`/api/queue/${id}`);
      return response.data;
    } catch (error) {
      console.error('Error deleting queue item:', error);
      throw error;
    }
  },
};

File: mobile/app/src/main/java/com/medistream/staff/queue/QueueService.kt
package com.medistream.staff.queue

import com.medistream.patient.queue.PatientQueueRequest
import com.medistream.patient.queue.QueueJoinResponse
import retrofit2.Response
import retrofit2.http.*

interface QueueService {
    @GET("api/queue")
    suspend fun getQueue(): Response<List<QueueItem>>

    @POST("api/queue/join")
    suspend fun joinQueue(@Body request: PatientQueueRequest): Response<QueueJoinResponse>

    @PUT("api/queue/{id}")
    suspend fun updateQueueItem(@Path("id") id: Long, @Body queueItem: QueueItem): Response<QueueItem>

    @DELETE("api/queue/{id}")
    suspend fun deleteQueueItem(@Path("id") id: Long): Response<Void>
}
